.github/workflows/ci-main.yml

```diff
@@ -332,106 +332,99 @@ jobs:
           # Clean up the output base for the next run
           sudo rm -rf /var/tmp/bazel-output
 
-  # Build & test the arm64-linux variant of pocket-ic-server and makes
-  # it available as a GHA artifact
-  # NOTE: uses 'namespace.so' runners
-  bazel-test-arm64-linux:
-    name: Bazel Test arm64-linux
-    runs-on: namespace-profile-arm64-linux # profile created in namespace console
+  # Build & upload the arm64-{linux,darwin} variant of the PocketIC server,
+  # build the entire codebase on arm64-darwin,
+  # and run tests tagged to run on all platforms.
+  # NOTE: uses 'namespace.so' runners.
+  bazel-test-arm64:
+    strategy:
+      matrix:
+        include:
+          - os: linux
+            profile: namespace-profile-arm64-linux # profile created in namespace console
+          - os: darwin
+            profile: namespace-profile-darwin # profile created in namespace console
+    name: Bazel Test arm64-${{ matrix.os }}
+    runs-on: ${{ matrix.profile }}
+    timeout-minutes: 120
     if: github.repository == 'dfinity/ic' # only run on public repo, not private since Namespace runners are not configured there, so these CI jobs get stuck otherwise.
     steps:
       - name: Set up Bazel cache
         run: |
           # Creates a bazelrc configuration fragment which tells bazel where the cache lives.
           nsc bazel cache setup --bazelrc=/tmp/bazel-cache.bazelrc
+
       - uses: actions/checkout@v4
+
+      - name: Set up zig cache
+        run: |
+          mkdir -p /tmp/zig-cache
```

Comment:
interesting that this wasn't required on linux before. I think we could add an `if: matrix.os == "darwin"`  here or even check if it's still necessary

Reply:
And it works!